spec.html

```diff
@@ -24448,11 +24877,106 @@ <h1>Static Semantics: ContainsArguments ( ): a Boolean</h1>
       </emu-alg>
     </emu-clause>
 
+    <emu-clause id="sec-class-definitions-static-semantics-methoddecorators">
+      <h1>Static Semantics: Method Decorators</h1>
+      <emu-grammar>
+        ClassElement : DecoratorList? MethodDefinition
+      </emu-grammar>
+      <ul>
+        <li>
+          It is a Syntax Error if |DecoratorList| is present and PropName of |MethodDefinition| is `"constructor"`.
+        </li>
+      </ul>
+    </emu-clause>
+
+    <emu-clause id="sec-applydecoratorsanddefinemethod" type="abstract operation">
+      <h1>
+        ApplyDecoratorsAndDefineMethod (
+          _homeObject_: an Object,
+          _methodDefinition_: a ClassElementDefinition Record,
+          _extraInitializers_: a List of function objects,
+          _isStatic_: a Boolean,
+        ): either a normal completion containing ~unused~, or an abrupt completion
+      </h1>
+      <dl class="header">
+      </dl>
+      <emu-alg>
+        1. Perform ? ApplyDecoratorsToElementDefinition(_homeObject_, _methodDefinition_, _extraInitializers_, _isStatic_).
+        1. Perform ? DefineMethodProperty(_homeObject_, _methodDefinition_, _isStatic_).
+        1. Return ~unused~.
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-createfieldinitializerfunction" type="abstract operation">
+      <h1>
+        CreateFieldInitializerFunction (
+          _homeObject_: an Object,
+          _propName_: a property key or Private Name,
+          _Initializer_: a Parse Node,
+        ): a function object
+      </h1>
+      <dl class="header">
+      </dl>
+      <emu-alg>
+        1. Let _formalParameterList_ be an instance of the production <emu-grammar>FormalParameters : [empty]</emu-grammar>.
+        1. Let _scope_ be the LexicalEnvironment of the running execution context.
+        1. Let _privateScope_ be the running execution context's PrivateEnvironment.
+        1. Let _sourceText_ be the empty sequence of Unicode code points.
+        1. Let _initializer_ be OrdinaryFunctionCreate(%Function.prototype%, _sourceText_, _formalParameterList_, _Initializer_, ~non-lexical-this~, _scope_, _privateScope_).
+        1. Perform MakeMethod(_initializer_, _homeObject_).
+        1. Set _initializer_.[[ClassFieldInitializerName]] to _propName_.
+        1. Return _initializer_.
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-makeautoaccessorgetter" type="abstract operation">
+      <h1>
+        MakeAutoAccessorGetter (
+          _homeObject_: an Object,
+          _name_: a property key or Private Name,
+          _privateStateName_: a Private Name,
+        ): a function object
+      </h1>
+      <dl class="header">
+      </dl>
+      <emu-alg>
+        1. Let _getterClosure_ be a new Abstract Closure with no parameters that captures _privateStateName_ and performs the following steps when called:
+          1. Let _o_ be the *this* value.
+          1. Return ? PrivateGet(_privateStateName_, _o_).
```

Comment:
```suggestion
          1. Return ? PrivateGet(_o_, _privateStateName_).
```

Reply:
Also, PrivateGet requires an Object, but I don't think it's guaranteed that `_o_` is an Object.

(And similarly for `_o_` in MakeAutoAccessorSetter.)